File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int wid = Integer.MIN_VALUE;
        int hei = Integer.MIN_VALUE;
        
        for(int[] target : sizes){
            int targetWid = target[0];
            int targetHei = target[1];
            if(targetWid <= targetHei){
                wid = Math.max(targetHei, wid);
                hei = Math.max(targetWid, hei);
            } else {
                wid = Math.max(targetWid, wid);
                hei = Math.max(targetHei, hei);
            }
            
        }
        
        int answer = wid * hei;
        return answer;
    }
}

File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let wid = Number.MIN_VALUE;
    let hei = Number.MIN_VALUE;
    
    sizes.forEach(item => {
        const targetWid = item[0];
        const targetHei = item[1];
        if(targetWid <= targetHei){
            wid = Math.max(targetHei, wid);
            hei = Math.max(targetWid, hei);
        } else {
            wid = Math.max(targetWid, wid);
            hei = Math.max(targetHei, hei);
        } 
    });
    const answer = wid * hei;
    return answer;
}